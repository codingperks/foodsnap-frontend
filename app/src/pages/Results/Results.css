.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

/*.App-header {*/
/*  min-height: 94vh;*/
/*  display: flex;*/
/*  flex-direction: column;*/
/*  align-items: center;*/
/*  justify-content: center;*/
/*  text-align: left;*/
/*  font-size: calc(10px + 2vmin);*/
/*  color: #798777;*/
/*}*/

table {
  width: 100%;
  margin-bottom: 20px;
}
th, td {
  padding: 10px; /* Increase the padding value to increase the space between the cells */
  border: 1px solid black;
  width: 50%;
}

/*caption {*/
/*  font-size: 24;*/
/*  font-weight: bold;*/
/*}*/

/*pre {*/
/*  font-family: "Open Sans Bold'", sans-serif;*/
/*  font-size: 14px;*/
/*  line-height: 1.5;*/
/*  !*color: #333;*!*/
/*  padding: 10px;*/
/*  overflow-x: auto;*/
/*  white-space: pre-wrap;*/
/*  font-weight: normal;*/
/*}*/

/*header-font {*/
/*  font-size: 16px;*/
/*}*/

h1 {
  font-family: 'Tilt Warp', sans-serif;
}