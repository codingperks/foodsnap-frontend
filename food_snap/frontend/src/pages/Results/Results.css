.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #233029;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #3d8fd3;
}