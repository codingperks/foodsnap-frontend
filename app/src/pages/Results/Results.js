import React from 'react'
import './Results.css'
import { useLocation, useNavigate } from 'react-router-dom'

function Results () {

  const location = useLocation()
  const navigate = useNavigate()
  if (location.state.responseData) {
    return (
        <div className="App">
          <header className="App-header">
            <h1>Results</h1>
            <h5>Type of food:&nbsp;{location.state.responseData.label}</h5>
            <h5>Weight: {location.state.responseData.weight} grams</h5>
            <h5>Nutritional value</h5>
            <div>
              <p>Calories: {location.state.responseData.nutrition.ENERC_KCAL} KCal</p>
              <p>Fat: {location.state.responseData.nutrition.FAT} grams</p>
              <p>Protein: {location.state.responseData.nutrition.PROCNT} grams</p>
            </div>
              <button onClick={() => navigate('/')}>Try Again</button>
          </header>
        </div>
    )
  } else {
    return (
        <div className="App">
          <header className="App-header">
            <h1>No Result Available</h1>
            <button onClick={() => navigate('/')}>Try Again</button>
          </header>
        </div>
    )
  }
}

export default Results
