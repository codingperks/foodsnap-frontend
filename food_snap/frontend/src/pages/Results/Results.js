import React, { useState } from 'react';
import logo from '../../resources/logo_512.png';
import './Results.css';

function Results() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>
          Results
        </h1>
        <img src={logo} className="App-logo" alt="logo" />
      </header>
    </div>
  );
}

export default Results;
