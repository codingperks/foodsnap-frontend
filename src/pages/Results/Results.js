import React from 'react'
import './Results.css'
import Navigation from '../../elements/Navigation/Navigation'

function Results () {
  return (
    <div className="App">
      <header className="App-header">
        <h1>
          Results placeholder
        </h1>
      </header>
      <Navigation />
    </div>
  )
}

export default Results
