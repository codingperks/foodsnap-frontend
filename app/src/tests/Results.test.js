import React from "react";
import {render} from "@testing-library/react";
import Results from "../pages/Results/Results";
import {BrowserRouter} from 'react-router-dom'
import "@testing-library/jest-dom/extend-expect";

describe('Home', () => {
    // Creates a snapshot of the current Results page - this test then compares any new builds to this snapshot
    test('Matches DOM Snapshot', () => {
        const {asFragment} = render(
            <BrowserRouter>
                <Results/>
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});