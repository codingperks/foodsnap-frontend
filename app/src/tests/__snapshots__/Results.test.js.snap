// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-auto items-center justify-center bg-[color:var(--bg-light)] dark:bg-[color:var(--bg-dark)] min-h-screen min-w-screen overflow-x-hidden overflow-y-auto overscroll-none App"
  >
    <header
      class="flex flex-col min-h-screen min-w-screen flex-grow font-sans mt-5 text-align App-header bg-[color:var(--bg-light)] dark:bg-[color:var(--bg-dark)] overflow-x-hidden overscroll-none min-w-screen min-h-screen"
    >
      <style>
        @import url(https://fonts.googleapis.com/css2?family=Space+Grotesk:wght@500&family=Tilt+Warp&display=swap);
      </style>
      <button
        class="fixed items-center justify-center top-1 right-0 z-10 text-[color:var(--sidebar-button-light)] dark:text-[color:var(--sidebar-button-dark)] bg-transparent rounded-lg text-sm px-5 py-5 mr-2 mb-2"
        type="button"
      >
        <svg
          class="w-7 h-7"
          fill="currentColor"
          stroke="currentColor"
          stroke-width="3"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
      <main
        class="fixed overflow-y-scroll overflow-hidden z-10 inset-0 transform ease-in-out transition-all delay-50 duration-500 opacity-0 translate-x-full"
        style="z-index: 50;"
      >
        <div
          class="absolute inset-0 bg-gray-500 opacity-75"
        />
        <div
          class="w-screen max-w-lg right-0 bg-[color:var(--sidebar-bg-light)] dark:bg-[color:var(--sidebar-bg-dark)] h-full w-full absolute right-0transform translate-x-full ease-in"
          style="z-index: 50;"
        >
          <div
            class="flex items-center justify-between m-5"
          >
            <h1
              class="text-xl font-semibold text-[color:var(--sidebar-h1-light)] dark:text-[color:var(--sidebar-h1-dark)]"
            >
              Welcome to FoodSnap!
            </h1>
            <button
              aria-label="Close"
              class="text-[color:var(--sidebar-p-light)] dark:text-[color:var(--sidebar-p-dark)] focus:outline-none"
            >
              <svg
                class="h-6 w-6 fill-current text-[color:var(--sidebar-button-light)] dark:text-[color:var(--sidebar-button-dark)]"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="heroicon-ui"
                  d="M6.7 5.3a1 1 0 0 0-1.4 1.4L10.6 12l-5.3 5.3a1 1 0 1 0 1.4 1.4L12 13.4l5.3 5.3a1 1 0 0 0 1.4-1.4L13.4 12l5.3-5.3a1 1 0 0 0-1.4-1.4L12 10.6l-5.3-5.3z"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex flex-col m-5"
          />
        </div>
        <div
          class=" w-screen h-full cursor-pointer z-20"
        />
      </main>
      <div
        class="h-screen flex flex-col absolute inset-0 align-middle place-items-center justify-center items-center"
      >
        <div
          class="flex -mt-20 place-items-center align-middle text-center text-3xl flex-col justify-center items-center text-[color:var(--h1-light)] dark:text-[color:var(--h1-dark)]"
        >
          <h1
            class="flex justify-center text-center text-4xl"
          >
            No Results Found
          </h1>
          <p
            class="flex mt-10 mx-5 font-sans text-xl"
          >
            We failed to identify a food item in your picture. Please make sure that the picture includes a food item, and that the picture is taken from above.
          </p>
        </div>
        <div
          class="justify-center"
        >
          <button
            class="mt-10 p-0.5 text-l font-medium text-[color:var(--button-text-light)] dark:text-[color:var(--button-text-dark)] rounded-lg group group-hover:from-pink-500 group-hover:to-orange-400 hover:text-[color:var(--button-text-light)] hover:dark:text-[color:var(--button-text-dark)]  transition-transform duration-300 ease-in-out transform hover:scale-105 active:scale-95"
          >
            <span
              class="px-8 font-bold text-lg py-2.5 transition-all ease-in duration-75 bg-[color:var(--button-background-light)] dark:bg-[color:var(--button-background-dark)] rounded-md group-hover:bg-opacity-0"
            >
              Try Again
            </span>
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
